device-types/Cisco/WS-C3850-48XS-E.yaml

```diff
@@ -0,0 +1,121 @@
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/4
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/5
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/6
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/7
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/8
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/9
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/10
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/11
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/12
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/13
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/14
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/15
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/16
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/17
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/18
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/19
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/20
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/21
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/22
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/23
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/24
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/25
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/26
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/27
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/28
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/29
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/30
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/31
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/32
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/33
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/34
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/35
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/36
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/37
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/38
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/39
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/40
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/41
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/42
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/43
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/44
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/45
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/46
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/47
+    type: 10gbase-x-sfpp
+  - name: TenGigabitEthernet1/0/48
+    type: 10gbase-x-sfpp
+module-bays:
+  - name: Network Module
+    position: '1'
+console-ports:
+  - name: con0
+    type: rj-45
+  - name: usb0
+    type: usb-mini-b
+power-ports:
+  - name: PSU0
```

Comment:
I'm working on making the changes you requested, but the scope of this PR has been steadily creeping larger and larger.

My original goal with this PR was to add devices which were missing from the library, adapting from similar devices which were already in the library. I will make the power port change you requested, but changing making all the fans of all these models modular is too much for me at the moment. The existing C3850 models in the library have fans, instead of fan modules, and so the devices i've added also have fans instead of fan modules. I do not have time to fix these pre-existing discrepancies right now.

Sorry.

Reply:
I have to admit I was surprised that you added the `cisco-` slug changes in this PR, I would have expected them to be made in another PR. That being said, I went through and reviewed all of the changes you had added.

Also, I was just mentioning that the fans which are modular should be added to the module-bays, not specifically that you need to make that change.. I am sorry if I wasnt clear about that.. It was more of a statement than a request. The PSU changes are the only important ones here

Reply:
I am sorry for the confusion regarding that.